src/infra/src/table/migration/m20241119_000001_create_dashboards_table.rs

```diff
@@ -0,0 +1,202 @@
+        // integer "snowflake".
+        .col(ColumnDef::new(Dashboards::DashboardId).string_len(256).not_null())
+        // Foreign key to the folders table.
+        .col(ColumnDef::new(Dashboards::FolderId).big_integer().not_null())
+        // Identifier of the user that owns the dashboard.
+        .col(ColumnDef::new(Dashboards::Owner).string_len(256).not_null())
+        .col(ColumnDef::new(Dashboards::Role).string_len(256).null())
+        .col(ColumnDef::new(Dashboards::Title).string_len(256).not_null())
+        .col(ColumnDef::new(Dashboards::Description).text().null())
+        .col(ColumnDef::new(Dashboards::Data).json().not_null())
+        .col(ColumnDef::new(Dashboards::Version).integer().not_null())
+        .col(
+            ColumnDef::new(Dashboards::CreatedAt)
+                .big_integer()
+                .not_null(),
+        )
+        .foreign_key(
+            sea_query::ForeignKey::create()
+                    .name(DASHBOARDS_FOLDERS_FK)
+                    .from(Dashboards::Table, Dashboards::FolderId)
+                    .to(Folders::Table, Folders::Id)
+        )
+        .to_owned()
+}
+
+/// Statement to create unique index on dashboard_id.
+fn create_dashboards_dashboard_id_idx_stmnt() -> IndexCreateStatement {
+    sea_query::Index::create()
+        .if_not_exists()
+        .name(DASHBOARDS_FOLDER_ID_DASHBOARD_ID_IDX)
+        .table(Dashboards::Table)
+        .col(Dashboards::FolderId)
+        .col(Dashboards::DashboardId)
+        .unique()
+        .to_owned()
+}
+
+/// Identifiers used in queries on the dashboards table.
+#[derive(DeriveIden)]
+enum Dashboards {
+    Table,
+    Id,
+    DashboardId,
+    FolderId,
+    Owner,
+    Role,
+    Title,
+    Description,
+    Data,
+    Version,
+    CreatedAt,
+}
+
+/// Identifiers used in queries on the folders table.
+#[derive(DeriveIden)]
+enum Folders {
+    Table,
+    Id,
+}
+
+#[cfg(test)]
+mod tests {
+    use collapse::*;
+
+    use super::*;
+
+    #[test]
+    fn postgres() {
+        collapsed_eq!(
+            &create_dashboards_table_statement().to_string(PostgresQueryBuilder),
+            r#"
+                CREATE TABLE IF NOT EXISTS "dashboards" ( 
+                "id" bigserial NOT NULL PRIMARY KEY, 
+                "dashboard_id" varchar(256) NOT NULL, 
+                "folder_id" bigint NOT NULL, 
+                "owner" varchar(256) NOT NULL, 
+                "role" varchar(256) NULL, 
+                "title" varchar(256) NOT NULL, 
+                "description" text, 
+                "data" json NOT NULL,
+                "version" integer NOT NULL, 
+                "created_at" bigint NOT NULL, 
+                CONSTRAINT "dashboards_folders_fk" FOREIGN KEY ("folder_id") REFERENCES "folders" ("id") 
+            )"#
+        );
+        assert_eq!(
+            &create_dashboards_folder_id_dashboard_id_idx_stmnt().to_string(PostgresQueryBuilder),
```

Comment:
this is probably meant to call this one insteadn? 
```suggestion
            &create_dashboards_dashboard_id_idx_stmnt().to_string(PostgresQueryBuilder),
```

Reply:
Yeah not sure what happened there. I thought I renamed that function 🤔 
fixed